baricadr/model/repos.py

```diff
@@ -157,42 +168,65 @@ def _get_freezable(self, path, remote_list, force=False):
+                    current_app.logger.info("Evaluating freezable for path: %s " % (candidate))
                     excluded = False
                     for ex in excludes:
                         if fnmatch.fnmatch(candidate, ex.strip()):
                             current_app.logger.info("Found excluded path: %s with expression %s" % (candidate, ex.strip()))
                             excluded = True
                             break
-                    if not excluded and (force or self._can_freeze(candidate)) and (self.relative_path(candidate) in remote_list):
+                    if not excluded and self._can_freeze(candidate, remote_list, force):
                         freezables.append(candidate)
 
         return freezables
 
-    def _can_freeze(self, file_to_check):
+    def _can_freeze(self, file_to_check, remote_list, force):
         """
         Check if a file should be freezed or not
 
-        :type path: str
-        :param path: Path of a file to check
+        :type file_to_check: str
+        :param file_to_check: Path of a file to check
+
+        :type force: bool
+        :param force: Whether to ignore atime
 
         :rtype: bool
         :return: True if the file should be freezed
         """
 
+        relative_path = self.relative_path(file_to_check)
+        # Check if in remote list
+        remote_file = next((item for item in remote_list if item["Path"] == relative_path), None)
+
+        if not remote_file:
+            return False
+
+        # Check if modified since pulled
+        tz = get_localzone()
+
+        last_modif_remote = dateutil.parser.isoparse(remote_file['ModTime'])
+        last_modif_local = datetime.datetime.fromtimestamp(os.stat(file_to_check).st_mtime, tz=tz)
+
+        if force:
+            current_app.logger.info("Checking if we should freeze '%s': local modification on '%s' , remote modification on '%s' => Delta is %s seconds" % (file_to_check, last_modif_local, last_modif_remote, (last_modif_local - last_modif_remote).total_seconds()))
+
+        # Assuming 10s delay? Maybe more? -> Might need to be fine-tuned. Tests shows 0.22s
+        if (last_modif_local - last_modif_remote).total_seconds() > 10:
+            return False
+
         last_access = datetime.datetime.fromtimestamp(os.stat(file_to_check).st_atime).date()
         now = datetime.date.today()
         delta = now - last_access
         delta = delta.days
-        current_app.logger.info("Checking if we should freeze '%s' (freeze_age=%s): last accessed on %s (%s days ago) =>  %s" % (file_to_check, self.freeze_age, last_access, delta, delta > self.freeze_age))
+        current_app.logger.info("Checking if we should freeze '%s' (freeze_age=%s): last accessed on %s (%s days ago) =>  %s" % (file_to_check, self.freeze_age, last_access, delta, (force or (delta > self.freeze_age))))
 
-        return delta > self.freeze_age
+        return force or (delta > self.freeze_age)
```

Comment:
If force is True, we could return earlier to avoid a useless check of the modification time?

Reply:
Sure